cmake/sysbuild/image_signing.cmake

```diff
@@ -137,6 +137,22 @@ function(zephyr_mcuboot_tasks)
     set(input ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_NAME})
   endif()
 
+  if(SB_CONFIG_SOC_SERIES_NRF54LX AND SB_CONFIG_BOOT_SIGNATURE_TYPE_ED25519)
+    # Enforce SHA512 on nrf54l with ED25519; no need to bring SHA256 since ED25519
+    # is using SHA512 internally anyway.
+    set(imgtool_sha "--sha 512")
+  else()
+    set(imgtool_sha)
+  endif()
+
+  if(SB_CONFIG_BOOT_SIGNATURE_TYPE_PURE)
+    set(imgtool_pure "--pure")
+  else()
+    set(imgtool_pure)
```

Comment:
just append to imgtool_extra, do not create new variables

Reply:
Did exactly that, but in Zephyr part.